front/src/modules/timesStops/hooks/useComputedMargins.tsx

```diff
@@ -0,0 +1,73 @@
+import type { SimulationResponseSuccess } from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainScheduleV2/utils';
+
+import { marginRegExValidation } from '../consts';
+import { formatDigitsAndUnit } from '../utils';
+
+let marginUnit = '%';
+
+const useComputedMargins = (
+  trainSimulation: SimulationResponseSuccess,
+  opPoint: OperationalPointWithTimeAndSpeed,
+  nextOpPoint: OperationalPointWithTimeAndSpeed,
+  selectedTrainSchedule: TrainScheduleResult,
+  pathId: string
+) => {
+  const theoreticalMarginIndex = selectedTrainSchedule?.margins?.boundaries?.findIndex(
+    (id) => id === pathId
+  );
+  if (!theoreticalMarginIndex) {
+    return ['', '', ''];
+  }
+  const theoreticalMargin = selectedTrainSchedule.margins!.values[theoreticalMarginIndex + 1];
+  const theoMarginRegexMatches = theoreticalMargin?.match(marginRegExValidation);
+  marginUnit = theoMarginRegexMatches?.[3] || marginUnit;
+  const opPointBaseTime = interpolateValue(trainSimulation.base, opPoint?.position * 1000, 'times');
+  const nextOpPointBaseTime = interpolateValue(
+    trainSimulation.base,
+    nextOpPoint?.position * 1000,
```

Comment:
Remove all the ? which are not necessary and use the utils function mToMm instead of `* 1000`
```suggestion
    nextOpPoint.position * 1000,
```

Reply:
done